pkg/controlplane/instance.go

```diff
@@ -482,9 +489,40 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		return nil
 	})
 
+	if utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.APIServerIdentity) {
+		m.GenericAPIServer.AddPostStartHookOrDie("start-kube-apiserver-lease-controller", func(hookContext genericapiserver.PostStartHookContext) error {
+			kubeClient, err := kubernetes.NewForConfig(hookContext.LoopbackClientConfig)
+			if err != nil {
+				return err
+			}
+			controller := lease.NewController(
+				clock.RealClock{},
+				kubeClient,
+				m.GenericAPIServer.APIServerID,
+				int32(c.ExtraConfig.IdentityLeaseDurationSeconds),
+				nil,
+				time.Duration(c.ExtraConfig.IdentityLeaseRenewIntervalSeconds)*time.Second,
+				metav1.NamespaceSystem,
+				labelAPIServerHeartbeat)
+			go controller.Run(wait.NeverStop)
+			return nil
+		})
+	}
+
 	return m, nil
 }
 
+func labelAPIServerHeartbeat(lease *coordinationapiv1.Lease) error {
+	if lease.Labels == nil {
+		lease.Labels = map[string]string{}
+	}
+	// This label indicates:
+	//   1. the lease object is a heartbeat lease (different from leader election leases)
+	//   2. the component sending the heartbeat is kube-apiserver
+	lease.Labels["k8s.io/component-heartbeat"] = "kube-apiserver"
```

Comment:
Do we have to use this label name? Can we do just "k8s.io/component"

Reply:
Also we should make this a string constant, probably.

Reply:
Have thought about using two labels: `k8s.io/lease-type=heartbeat` and `k8s.io/component=kube-apiserver`. Thought it was too verbose. 

We can use `k8s.io/component`. I've verified that currently there is nothing using it in-tree.